app/gui2/src/stores/graph/index.ts

```diff
@@ -186,11 +187,41 @@ export const useGraphStore = defineStore('graph', () => {
     }
   }
 
-  function methodAstInModule(mod: Ast.Module) {
-    const topLevel = mod.root()
-    if (!topLevel) return
+  function getExecutedMethodAst(module?: Ast.Module): Result<Ast.Function> {
+    const executionStackTop = proj.executionContext.getStackTop()
+    switch (executionStackTop.type) {
+      case 'ExplicitCall': {
+        return getMethodAst(executionStackTop.methodPointer, module)
+      }
+      case 'LocalCall': {
+        const exprId = executionStackTop.expressionId
+        const info = db.getExpressionInfo(exprId)
+        const ptr = info?.methodCall?.methodPointer
+        if (!ptr) return Err("Unknown method pointer of execution stack's top frame")
+        return getMethodAst(ptr, module)
+      }
+    }
+  }
+
+  function getMethodAst(ptr: MethodPointer, module?: Ast.Module): Result<Ast.Function> {
+    const topLevel = (module ?? syncModule.value)?.root()
```

Comment:
Instead of using a fallback here, I think the `module` argument should be required and `syncModule.value` should simply be used as a parameter when it's appropriate (i.e. in `computed` above).

Reply:
This `module` cannot be just any module - it should always be a `syncModule` or one of its edited states. I'd rather rename parameter name to `edit` and keep it optional. It's published as method of "graph" storage, so it should be obvious of what module we want to take a definition.